enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/ClientEvaluationService.java

```diff
@@ -103,33 +98,49 @@ public void initClientEvaluation(final OwnerId ownerId, final IdentityVerificati
     public void processClientEvaluation(final IdentityVerificationEntity identityVerification, final OwnerId ownerId) {
         logger.debug("Client evaluation started for {}", identityVerification);
 
+        final String verificationId;
+        try {
+            verificationId = getVerificationId(identityVerification);
+        } catch (Exception e) {
+            processVerificationIdError(identityVerification, ownerId, e);
+            return;
+        }
+
         final EvaluateClientRequest request = EvaluateClientRequest.builder()
                 .processId(identityVerification.getProcessId())
                 .userId(identityVerification.getUserId())
                 .identityVerificationId(identityVerification.getId())
-                .verificationId(getVerificationId(identityVerification))
+                .verificationId(verificationId)
                 .build();
 
         final int maxFailedAttempts = config.getClientEvaluationMaxFailedAttempts();
         for (int i = 0; i < maxFailedAttempts; i++) {
             try {
                 final EvaluateClientResponse response = onboardingProvider.evaluateClient(request);
                 processEvaluationSuccess(identityVerification, ownerId, response);
-                break;
-            } catch (Throwable t) {
-                processEvaluationError(identityVerification, ownerId, t);
```

Comment:
When we really want to allow more attempts, we must not fail the identity verification after the first error. Covered by test since now.